### ISSUE TITLE
Questions about the test result of reproduction

### ISSUE THREAD

**Issue by A11en4z** [zhangce01/HiKER-SGG#16]
Hello, I am very interested in your excellent work and have tried to reproduce it.
I modified only the necessary paths, made no other changes to the parameters, and trained on a single GPU 3090.
I have a little problem with the results of the predcls task：

My best validation result in Log occurs at epoch11.

![best_mr_c_epoch11](https://github.com/user-attachments/assets/c987a7fc-5d43-4805-be58-02540ce51943)

However, when I use test.ipynb  and  load the corresponding confusion matrix, the "predcls  mean recall with constraint" drops to :

![predcls_test_epoch11](https://github.com/user-attachments/assets/b7cc90bb-e394-4f3c-ac75-74a26855f096)
I'm wondering if this is normal or if there's something wrong with my config because I've tried a few other checkpoints and still don't get better results.

**Comment by zhangce01** [comment]
Thank you for your interest in our work! It seems that you've achieved reasonable results on the validation set, but I suspect that your model may be overfitting. The small validation set (~5000 images) might not adequately reflect this issue. To investigate further, although it’s not considered best practice, you could test your model from earlier epochs to see if they produce better results, even with lower validation accuracy (Note that you should also save and load the  corresponding confusion matrix). Based on our experiments, you should consistently achieve at least 40% mR@100 without constraints, and with some randomness, the results could potentially be even higher.

**Comment by A11en4z** [comment]
> Thank you for your interest in our work! It seems that you've achieved reasonable results on the validation set, but I suspect that your model may be overfitting. The small validation set (~5000 images) might not adequately reflect this issue. To investigate further, although it’s not considered best practice, you could test your model from earlier epochs to see if they produce better results, even with lower validation accuracy (Note that you should also save and load the corresponding confusion matrix). Based on our experiments, you should consistently achieve at least 40% mR@100 without constraints, and with some randomness, the results could potentially be even higher.

Hello! I tried all the epoch results, but the best result is also about three points lower than the paper result. I'm wondering if there's something wrong with the training setup because earlier checkpoints didn't give better results in the test, and I noticed that you have an epoch of 30 in the paper but 20 in the code, does that make a difference?

**Comment by zhangce01** [comment]
I don't think it will make much difference, as the optimal model should be reached within 20 epochs. You could try adjusting (specifically decreasing) the initial learning rate slightly, or applying exponential learning rate decay, as these might have some influence on the performance. By the way, have you encountered any issues with numerical overflow, or any unexpected errors/warnings during your training process?

**Comment by A11en4z** [comment]
> I don't think it will make much difference, as the optimal model should be reached within 20 epochs. You could try adjusting (specifically decreasing) the initial learning rate slightly, or applying exponential learning rate decay, as these might have some influence on the performance. By the way, have you encountered any issues with numerical overflow, or any unexpected errors/warnings during your training process?

Thank you for your reply! That's the only warning I've encountered and I'm now trying to retrain and hope for better results.
![image](https://github.com/user-attachments/assets/3c61dbb5-2cc5-42f5-bf88-13e4d9ef6b7d)

**Comment by Nyeilim** [comment]
I can't reproduce the result too, also three points lower than the paper's. could you update corresponding ckpt file in your paper, if possible? I wanna train from that checkpoint. @zhangce01
